### PR TITLE
Move helper methods for adding webhooks to extension methods

### PR DESCRIPTION
**Pull request by ronaldbarendse**
### Prerequisites

- [x] I have added steps to test this contribution in the description below

### Description
This moves the helper methods for adding webhooks that were introduced in PR https://github.com/umbraco/Umbraco-CMS/pull/15161 to extension methods, so the `WebhookEventCollectionBuilder` class only contains the required logic for configuring the collection. I've also cleaned up this class, specifically by using `TryAddEnumerable()` to add the webhook event/notification type to the service collection and only calling `TryGetGenericArguments()` once (`IsOfGenericType()` is doing the same, but without returning the generic types).

I've also renamed `AddAllAvailableWebhooks()` to `AddCmsWebhooks()`, because 'all available' could suggest it does a type scan of all `IWebhookEvent` types and adds those, but instead it only adds the CMS supported ones.

Another thing I noticed is the lack of `DocumentType*` events: do we want to add those as well? And if PR https://github.com/umbraco/Umbraco-CMS/pull/15337 gets merged, we should ensure the `AddHealthCheckWebhooks()` helper methods gets moved into an extension method too 😄

Testing can be done by adding the following composer, which removes the default events and adds all content ones instead:

```c#
using Umbraco.Cms.Core.Composing;

public class WebhookComposer : IComposer
{
    public void Compose(IUmbracoBuilder builder)
        => builder.WebhookEvents()
            .Clear()
            .AddContentWebhooks();
}
```

Or if you want all CMS supported ones added (ignoring the duplicate default ones), try:
```c#
using Umbraco.Cms.Core.Composing;

public class WebhookComposer : IComposer
{
    public void Compose(IUmbracoBuilder builder)
        => builder.WebhookEvents().AddCmsWebhooks();
}
```

I'd also recommend a sanity check to see whether the webhooks get fired, since this does contain a change in the registration of the notification handlers.

### CODE REVIEW

File: src/Umbraco.Core/Webhooks/WebhookEventCollectionBuilderExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+        => builder
+            .Append<MediaDeletedWebhookEvent>()
+            .Append<MediaSavedWebhookEvent>()
+            .Append<MediaEmptiedRecycleBinWebhookEvent>()
+            .Append<MediaMovedWebhookEvent>()
+            .Append<MediaMovedToRecycleBinWebhookEvent>()
+            .Append<MediaTypeChangedWebhookEvent>()
+            .Append<MediaTypeDeletedWebhookEvent>()
+            .Append<MediaTypeMovedWebhookEvent>()
+            .Append<MediaTypeSavedWebhookEvent>();
```
Comment (ronaldbarendse) [inline src/Umbraco.Core/Webhooks/WebhookEventCollectionBuilderExtensions.cs:146]: Should the media type events move to their own `AddMediaTypeWebhooks()`, similar to `AddMemberTypeWebhooks()`?

**Comment by bjarnef** [comment]
I agree `AddCmsWebhooks()` is a better name, I considered `AddCoreWebhooks()`, but I think `Cms` is more clear - I guess there could be a `AddFormWebhooks()` in Umbraco Forms for instance.

File: src/Umbraco.Core/Webhooks/WebhookEventCollectionBuilderExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+            .Append<TemplateSavedWebhookEvent>();
+
+    /// <summary>
+    /// Adds the user webhook events.
+    /// </summary>
+    /// <param name="builder">The builder.</param>
+    /// <returns>
+    /// The builder.
+    /// </returns>
+    public static WebhookEventCollectionBuilder AddUserWebhooks(this WebhookEventCollectionBuilder builder)
```
[the remark this answers is not in the document]
Comment (Zeegaan) [reply src/Umbraco.Core/Webhooks/WebhookEventCollectionBuilderExtensions.cs:261]: Yea I think splitting these would be nice, I'd rather have more small ones than one big 🙈

File: src/Umbraco.Core/Webhooks/WebhookEventCollectionBuilderExtensions.cs (hunk 1 above)
Comment (Zeegaan) [reply src/Umbraco.Core/Webhooks/WebhookEventCollectionBuilderExtensions.cs:146]: I think `types` should move to their own as well 👍

File: src/Umbraco.Core/Webhooks/WebhookEventCollectionBuilderExtensions.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (ronaldbarendse) [reply src/Umbraco.Core/Webhooks/WebhookEventCollectionBuilderExtensions.cs:261]: And this is where the fluent API in PR https://github.com/umbraco/Umbraco-CMS/pull/15345 makes even more sense! Adding all user events can be done using `builder.WebhookEvents().AddCms(x => x.AddUser());`, only the default user saved/deleted events using `builder.WebhookEvents().AddCms(x => x.AddUser(onlyDefault: true));` and just user login events using `builder.WebhookEvents().AddCms(x => x.AddUser(x => x.AddLogin()));` 🙌🏻

**Pull request merged**